File: src/leetcode/java/RansomNote_383.java
package leetcode.java;

public class RansomNote_383 {
  public boolean canConstruct(String ransomNote, String magazine) {
    int[] arr = new int[128];
    for (char c : magazine.toCharArray()) arr[c]++;
    for (char c : ransomNote.toCharArray()) {
      if (--arr[c] < 0) return false;
    }
    return true;
  }
}

File: src/leetcode/python/RansomNote_383.py
from collections import Counter


class RansomNote_383(object):
    def canConstruct(self, ransomNote, magazine):
        """
        :type ransomNote: str
        :type magazine: str
        :rtype: bool
        """
        maz = Counter(magazine)
        for c in ransomNote:
            if c not in maz or maz[c] <= 0:
                return False
            maz[c] -= 1
        return True
